### PR TITLE
Advanced MSBuild project settings

### PR DESCRIPTION
**Pull request by Fireboyd78**
I've modified the project/solution settings to behave more dynamically and allow the use of either Visual Studio 2013 or 2015 without the need of updating the solution/project files. If you're using VS2015, it'll compile with the v140 platform toolset -- otherwise it'll use v120. I made Visual Studio 2013 the minimum version required to open the solution, so hopefully there's no need to put in place a "fallback" value for this. If you feel this should be handled differently, let me know.

I've also added an MSBuild Target that checks to make sure the user has installed all necessary SDK files before trying to generate any code (very useful and prevents a huge chain of compiler errors). This can be located at the end of the .vcxproj file.

I'm fairly experienced with MSBuild stuff so if you have any questions/comments/concerns, feel free to let me know!

### CODE REVIEW

**Comment by crosire** [comment]
Nice! It's been a while since I last digged into MSBuild (wrote a Emscripten toolset including compiler/linker properties and all that). The "VerifySDKFiles" target is really useful.

File: ScriptHookVDotNet.sln [hunk 1]
```diff
@@ -1,8 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.40629.0
-MinimumVisualStudioVersion = 10.0.40219.1
+# Visual Studio 2013 or higher
```
Comment (crosire) [inline ScriptHookVDotNet.sln]: This line is responsible for the solution icon displayed in the explorer. "Visual Studio 2013" adds a "12" badge, "Visual Studio 14" adds "14" badge. Unfortunately, this one removes that, so I'd remove the " or higher" part.

Comment (Fireboyd78) [inline ScriptHookVDotNet.sln]: Guess that's what I get for assuming it was just a comment, lol. I'll fix it tomorrow when I get a chance. Ideally, what should it be? 2013 or 14?

Comment (crosire) [inline ScriptHookVDotNet.sln]: I'd go with 2013, since that's the minimum.

**Comment by Fireboyd78** [comment]
Alright I finally got around to fixing it, sorry about that!

**Pull request merged**